src/main.go

```diff
@@ -0,0 +1,241 @@
+					Commands: map[string]cli.CommandFactory{
+						"workflownotification": func() cli.Command {
+							return &WorkflowNotificationCommand{}
+						},
+					},
+				}
+			},
+		},
+	}
+}
+
+func (c *WorkflowNotificationCommand) Flags() *cli.FlagSet {
+	set := cli.NewFlagSet()
+
+	f := set.NewSection("Chat space options")
+
+	f.StringVar(&cli.StringVar{
+		Name:    "webhook-url",
+		Example: "https://chat.goog...",
+		Default: "",
+		Target:  &c.flagWebhookUrl,
+		Usage:   `Webhook URL from google chat`,
+	})
+
+	return set
+}
+
+func (c *WorkflowNotificationCommand) Run(ctx context.Context, args []string) error {
+	f := c.Flags()
+	if err := f.Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	args = f.Args()
+	if len(args) != 0 {
+		return fmt.Errorf("expected 0 arguments, got %q", args)
+	}
+
+	ghJsonStr := os.Getenv(githubContextEnv)
+	if ghJsonStr == "" {
+		return fmt.Errorf("environment var %s not set", githubContextEnv)
+	}
+	jobJsonStr := os.Getenv(jobContextEnv)
+	if jobJsonStr == "" {
+		return fmt.Errorf("environment var %s not set", jobContextEnv)
+	}
+
+	ghJson := map[string]any{}
+	jobJson := map[string]any{}
+	if err := json.Unmarshal([]byte(ghJsonStr), &ghJson); err != nil {
+		return fmt.Errorf("failed unmarshaling %s: %w", githubContextEnv, err)
+	}
+	if err := json.Unmarshal([]byte(jobJsonStr), &jobJson); err != nil {
+		return fmt.Errorf("failed unmarshaling %s: %w", jobContextEnv, err)
+	}
+
+	b, err := generateMessageBody(ghJson, jobJson, time.Now())
+	if err != nil {
+		return fmt.Errorf("failed to generate message body: %w", err)
+	}
+
+	url := c.flagWebhookUrl
+
+	request, err := http.NewRequestWithContext(ctx, "POST", url, bytes.NewBuffer(b))
+	if err != nil {
+		return fmt.Errorf("creating http request failed: %w", err)
+	}
+
+	client := http.Client{}
+	resp, err := client.Do(request)
+	if err != nil {
+		return fmt.Errorf("sending http request failed: %w", err)
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("unexpected HTTP status code %d (%s)", resp.StatusCode, http.StatusText(resp.StatusCode))
+	}
+
+	return nil
+}
+
+func main() {
+	ctx, done := signal.NotifyContext(context.Background(),
+		syscall.SIGINT, syscall.SIGTERM)
+	defer done()
+
+	if err := realMain(ctx); err != nil {
+		done()
+		fmt.Fprintln(os.Stderr, err.Error())
+		os.Exit(1)
+	}
+}
+
+func realMain(ctx context.Context) error {
+	return rootCmd().Run(ctx, os.Args[1:]) //nolint:wrapcheck // Want passthrough
+}
+
+func generateMessageBody(ghJson, jobJson map[string]any, timestamp time.Time) ([]byte, error) {
+	timezoneLoc, _ := time.LoadLocation("America/Los_Angeles")
+
+	var iconUrl string
+	switch jobJson["status"] {
+	case "success":
+		iconUrl = "https://github.githubassets.com/favicons/favicon.png"
+	default:
+		iconUrl = "https://github.githubassets.com/favicons/favicon-failure.png"
+	}
+
+	jsonData := map[string]any{
```

Comment:
Should we generalize this API to allow users to provide their own title, subtitle and body?

I see more use cases then a GitHub job status indicator.

Reply:
I figured that would be a good follow-up

Reply:
+1

Reply:
Yes, that was on the plan for future iterations.